build.gradle

```diff
@@ -355,30 +290,69 @@ configure(javaProjects) {
         }
     }
 
-    task relocateSourceReferencesToInternalDependencies(type: ShadowJar) {
-        destinationDir = jar.destinationDir
-        archiveName = jar.baseName + '-relocated.jar'
+    // Create the shaded JAR.
+    task shadedJar(type: ShadowJar,
+                   group: 'Build',
+                   description: 'Builds the shaded JAR.') {
+        dependsOn tasks.classes
```

Comment:
> I believe ShadowJar handles this based on the configurations setting, but not sure.

It does when we use it as an extension. We here uses it just as a task, so we need to configure it manually.

> If we need it, I think it's more idiomatic to put into the task function arguments.

What is 'putting into the task function arguments'?